File: engine/model/src/main/scala/tools/packet/showitemgaininchat/ShowItemGainInChat.scala
package tools.packet.showitemgaininchat

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowItemGainInChat(private var _itemId: Int, private var _quantity: Short) extends PacketInput {
  def itemId: Int = _itemId

  def quantity: Short = _quantity

  override def opcode(): SendOpcode = SendOpcode.SHOW_ITEM_GAIN_INCHAT
}

File: engine/engine-model/src/main/java/tools/packet/showitemgaininchat/ShowItemGainInChat.java
package tools.packet.showitemgaininchat;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowItemGainInChat(Integer itemId, Short quantity) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_ITEM_GAIN_IN_CHAT;
   }
}